website/docs/reference/appsmith-framework/query-object.md

```diff
@@ -3,91 +3,119 @@ sidebar_position: 3
+| ------------- | --------------------------------------------------------------------------------------------------- |
+| **params**    | An object containing key-value pairs to pass into the query. Accessed with `{{ this.params.key }}`. |
+
+
+This function returns a JavaScript **promise**, which can be used to handle async actions in sequence. Use `.then()` and `.catch()` to write code to be executed when the query returns successfully or in error, respectively. Or, use `async/await` syntax.
 
 ```javascript
-run(onSuccess: Function, onError: Function, params: Object): void
+// Using promise syntax to chain actions in sequence
+{{
+    Query1.run(params)
+        .then(() => {...}) // run after the query is successful
+        .catch(() => {...}) // run if the query encounters any errors
+}}
 ```
 
-where **onSuccess** and **onError** are functions and **params** is a dictionary of key-value pairs.
-
-:::info
-We suggest you use the JavaScript Promise signature as it makes the code easy and readable. Callbacks are an old way and will be deprecated soon.
+To learn more about chaining actions to create complex workflows, see [complex workflows](/core-concepts/writing-code/workflows#complex-workflows).
 
-We recommend using the following Promise syntax:
+### Pass parameters to `run()`
 
-```
-Query.run(params)
-    .then(() => {...}) // run after the query is successful
-    .catch(() => {...}) // run if the query encounters any errors
-```
-:::
+When writing queries, you typically interpolate data from widgets or the [Appsmith Store](/reference/appsmith-framework/widget-actions/store-value) directly into the query body. Sometimes though, it's useful to pass non-widget data into the queries programmatically, using the same query multiple times in a loop.
 
-#### Arguments
+For this, use the `params` argument to pass an object of key-value pairs into your query. You can access these values within the query using `{{ this.params.key }}`.
 
-| Argument Name | Description                                                                                  |
-| ------------- | -------------------------------------------------------------------------------------------- |
-| **params**    | An object containing key-value pairs to pass into the query, accessed with `this.params.key` |
-| **onSuccess** | The function to be executed when the run method succeeds                                     |
-| **onError**   | The function to be executed when the run method fails                                        |
+#### Example use
 
-### Passing Params to Run
+Imagine that you have an API endpoint that returns data about teams of employees. `api.domain/teams/1` returns Team 1, `api.domain/teams/2` returns Team 2, etc. It would be useful to write a function that uses a single button click to return aggregated data from a number of selected teams.
```

Comment:
I agree it turned out way longer/more complex than I initially intended; I did keep it for the time being b/c I thought it might be useful for turning into a how-to guide or something in the future.

Basically I sank too much time into it and now I'm attached 🙃 

I'll replace it but I'll keep the example locally, maybe hang it on my refrigerator or something